### PR TITLE
Adding error enum to OnAccountChanged event and emitting as needed

### PR DESCRIPTION
**Pull request by tonyr59h**

### CODE REVIEW

**Comment by maxme** [comment]
Renaming done.

File: fluxc/src/main/java/org/wordpress/android/fluxc/store/AccountStore.java [hunk 1]
```diff
@@ -141,6 +143,24 @@ public static AuthenticationError fromString(String string) {
         }
     }
 
+    public enum AccountError {
+        ACCOUNT_FETCH_ERROR,
+        SETTINGS_FETCH_ERROR,
+        SETTINGS_POST_ERROR,
+        GENERIC_ERROR;
+
+        public static AccountError fromString(String string) {
```
Comment (maxme) [inline fluxc/src/main/java/org/wordpress/android/fluxc/store/AccountStore.java]: We don't need this method in `AccountError`, we don't read the error from the API.

File: fluxc/src/main/java/org/wordpress/android/fluxc/store/AccountStore.java [hunk 2]
```diff
@@ -239,12 +259,16 @@ public void onAction(Action action) {
             if (!checkError(data, "Error fetching Account via REST API (/me)")) {
                 mAccount.copyAccountAttributes(data.account);
                 updateDefaultAccount(mAccount, AccountAction.FETCH_ACCOUNT);
+            } else {
+                emitAccountChangeError(AccountError.ACCOUNT_FETCH_ERROR);
```
Comment (maxme) [inline fluxc/src/main/java/org/wordpress/android/fluxc/store/AccountStore.java]: I wonder if we should attach more info the the event. I think that's ok for a first pass. We could revisit later, we also have to manage generic errors somehow (timeout, network error).

Comment (tonyr59h) [inline fluxc/src/main/java/org/wordpress/android/fluxc/store/AccountStore.java]: I agree that there should be more info with these sorts of events but with our error handling in flux at the moment (four open issues to improve error handling) this should do for now.

**Comment by maxme** [comment]
:shipit:

**Pull request merged**